vacationtime-solution/src/Code.js

```diff
@@ -0,0 +1,195 @@
+      lastRow - frozenRows, 1);
+  notifiedColRange.setDataValidation(rule);
+  notifiedColRange.setValue('NOT NOTIFIED'); 
+  return notifiedColRange;
+}
+
+/**
+ * Creates a calendar event for an employee whose vacation
+ * request has been approved.
+ * @param {String} managerEmail Manager's email.
+ * @param {String} employeeName Name of employee.
+ * @param {String} employeeEmail Email of employee.
+ * @param {date} startDate Vacation request start date.
+ * @param {date} endDate Vacation request end date.
+**/
+function createCalEvent(managerEmail, employeeName, employeeEmail,
+                       startDate, endDate) { 
+    var managerCal = CalendarApp.getCalendarById(managerEmail);
+    // Creates a calendar event.
+    var event = managerCal.createEvent('APPROVED VACATION TIME FOR ' +
+        employeeName, startDate, endDate, {
+        description: 'Your vacation time from ' +
+                                            startDate + ' to ' + endDate +
+                                            ' has been approved! Enjoy!',
+        guests: employeeEmail,
+        sendInvites: true,
+    });
+}
+
+/**
+ * Sends e mails to employees whose vacation time request
+ * was NOT approved.
+ * @param {String} employeeEmail Email of employee.
+**/
+function sendEmails(employeeEmail) {
+  MailApp.sendEmail(employeeEmail, REJECTION_EMAIL_SUBJECT, 
+                        REJECTION_EMAIL_MESSAGE);
+}
+
+/**
+ * Checks the approval status of each employee and notifies
+ * them of their status accordingly.
+ * @param {String} employeeEmail Email of employee.
+ * @param {String} managerEmail Manager's email.
+ * @param {String} employeeName Name of employee.
+ * @param {String} approvalStatus Manager-set status.
+ * @param {date} startDate Vacation request start date.
+ * @param {date} endDate Vacation request end date.
+**/
+function checkApproval(employeeEmail, managerEmail, employeeName, 
+                       approvalStatus, startDate, endDate) {
+    var sheet = SpreadsheetApp.getActiveSheet();
+    var managerCal = CalendarApp.getCalendarById(managerEmail);
+    
+    // Checks approval status.
+    if (approvalStatus == 'NOT APPROVED') {
+      // Sends email of disapproval.
+      sendEmails(employeeEmail);
+    } else if (approvalStatus == 'APPROVED') {
+      // Creates calendar event.
+      createCalEvent(managerEmail, employeeName, employeeEmail,
+                      startDate, endDate);
+    }
+}
+
+/**
+ * Checks the notification status of each employee and, if not notified,
+ * notifies them of their status accordingly, through use of helper 
+ * functions.
+ */
+function notifyEmployees() {
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastRow = sheet.getLastRow();
+  var lastCol = sheet.getLastColumn();
+  var frozenRows = sheet.getFrozenRows();
+  var startRow = frozenRows + 1;
+  var numRows = lastRow - frozenRows;
+
+  // Gets  notified column values.
+  var notifiedColRange = createNotifiedColumn();
```

Comment:
Not sure this belongs here. Doing it here means the solution is effectively only good for one use. Otherwise every time you approve vacations you end up inserting a new notified column and send emails to everyone again (since the new column would be empty, previously notified employees are notified again.)